hdfs/src/main/scala/akka/stream/alpakka/hdfs/impl/HdfsFlowStage.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.hdfs.impl
+
+import akka.NotUsed
+import akka.event.Logging
+import akka.stream.alpakka.hdfs._
+import akka.stream.alpakka.hdfs.impl.HdfsFlowLogic.{FlowState, FlowStep, LogicState}
+import akka.stream.alpakka.hdfs.impl.strategy.DefaultRotationStrategy.TimeRotationStrategy
+import akka.stream.alpakka.hdfs.impl.writer.HdfsWriter
+import akka.stream.stage._
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import cats.data.State
+
+/**
+ * Internal API
+ */
+@akka.annotation.InternalApi
+private[hdfs] final class HdfsFlowStage[W, I, C](
+    ss: SyncStrategy,
+    rs: RotationStrategy,
+    settings: HdfsWritingSettings,
+    hdfsWriter: HdfsWriter[W, I]
+) extends GraphStage[FlowShape[HdfsWriteMessage[I, C], OutgoingMessage[C]]] {
+
+  private val in = Inlet[HdfsWriteMessage[I, C]](Logging.simpleName(this) + ".in")
+  private val out = Outlet[OutgoingMessage[C]](Logging.simpleName(this) + ".out")
+
+  override val shape: FlowShape[HdfsWriteMessage[I, C], OutgoingMessage[C]] = FlowShape(in, out)
+
```

Comment:
As the underlying writing abstraction is `java.io.OutputStream` which is blocking, we have to signal to the Akka Stream materializer that it should materialize HdfsFlowStage to a separate dispatcher. This allows other parts of the stream to continue uninpacted when this stage is going to block a thread during writing operations.

Therefore add the following here:
```
override def initialAttributes: Attributes =
  super.initialAttributes and ActorAttributes.IODispatcher
```

Reply:
Thank you for this comment. While I was running it, i was also profiling the app. I guess this is a nice improvement.